runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -28,15 +28,40 @@
 #include "chpl-comm.h"
 #include "chpl-mem.h"
 #include "chplmemtrack.h"
+#include "chplsys.h"
+#include "chpl-topo.h"
 #include "chpltypes.h"
 #include "error.h"
 
-static struct shared_heap {
+
+//
+// Information about shared heaps gotten from the comm layer.
+//
+typedef struct shared_heap {
   void* base;
   size_t size;
   size_t cur_offset;
   pthread_mutex_t alloc_lock;
-} heap; // static, will be "zero" initialized automatically
+} shared_heap_t;
+
+static shared_heap_t* heaps;
+static int num_heaps;
+
+static inline int get_num_heaps(void) {
+  return num_heaps;
```

Comment:
Ok. We could also just have get_num_heaps return chpl_topo_getNumNumaDomains(). get_num_heaps and get_heaps are't going to be used in performance critical areas. It's called when we're doing a chunk_allocation, but that doesn't get called very often.